src/Sarif/Baseline/ResultMatching/DataStructures/ExtractedResult.cs

```diff
@@ -15,7 +12,65 @@ namespace Microsoft.CodeAnalysis.Sarif.Baseline.ResultMatching
     public class ExtractedResult
     {
         public Result Result { get; set; }
-        
         public Run OriginalRun { get; set; }
+        public string RuleId { get; set; }
+
+        public ExtractedResult(Result result, Run run)
+        {
+            Result = result;
+            OriginalRun = run;
+
+            // Look up and cache the RuleId
+            RuleId = result.ResolvedRuleId(run);
+        }
+
+        /// <summary>
+        ///  Match the 'Category' of two ExtractedResults (Tool and RuleId)
+        /// </summary>
+        /// <param name="other">ExtractedResult to match</param>
+        /// <returns>True if Category is identical, False otherwise</returns>
+        public bool MatchesCategory(ExtractedResult other)
+        {
+            return this.RuleId == other.RuleId;
```

Comment:
>return [](start = 12, length = 6)

Is this also supposed to check the Tool?

Reply:
Tool is part of the category, but SarifLogMatcher only compares runs which have the same tool, so it's intentionally omitted. I made the comment clearer.